### ISSUE TITLE
AMD RX 7x00 vkAllocateMemory failed

### ISSUE THREAD

**Issue by NikAWing** [upscayl/upscayl#702]
### Checklist

- [X] I have **tried ALL** of the steps mentioned in the [troubleshooting guide](https://github.com/upscayl/upscayl/wiki/Troubleshooting).
- [X] I have **searched** for my issue in the [issues tab](https://github.com/upscayl/upscayl/issues?q=).
- [ ] My error involves vkEnumeratePhysicalDevices or vkCreateInstance, which means I should ignore the other issues and file this one despite the last checkbox.

### Describe the Bug

I just replaced my RX580 card with a new RX 7900 XT. Newest drivers, too.
I set the GPU ID to 0, try to upscale any photo, but all I get is error: vkAllocateMemory failed -13 and right after this as 2nd message I get error: vkQueueSubmit failed -2

### To Reproduce

load pic, try to upscale

### Upscayl Version (or commit hash)

2.9.8

### Platform

Windows

### OS Version

Win10

### GPU Name

RX 7900 XT

### Expected Behavior

no error and an upscaled picture :)

### Logs

```sh
🖼 Setting image path:  C:\!!pictmp\track.PNG
🗂 Setting output path:  C:\!!pictmp
🖼 imagePath:  C:\!!pictmp\track.PNG
🔤 Extension:  png
🔄 Resetting Upscaled Image Path
🏁 UPSCAYL
🐞 BACKEND REPORTED:  🖼️ Updating No Image Processing:  false
🐞 BACKEND REPORTED:  📐 Updating Compression:  0
🐞 BACKEND REPORTED:  Is Default Model? :  true
🐞 BACKEND REPORTED:  ✅ Upscayl Variables:  {"model":"realesrgan-x4plus","gpuId":"0","saveImageAs":"png","inputDir":"C:\\!!pictmp","outputDir":"C:\\!!pictmp","fullfileName":"track.PNG","fileName":"track","initialScale":"4","desiredScale":"4","outFile":"C:\\!!pictmp\\track_upscayl_4x_realesrgan-x4plus.png","compression":0}
🐞 BACKEND REPORTED:  📢 Upscayl Command:  -i,C:\!!pictmp\track.PNG,-o,C:\!!pictmp\track_upscayl_4x_realesrgan-x4plus.png,-s,4,-m,C:\Program Files\Upscayl\resources\models,-n,realesrgan-x4plus,-g,0,-f,png
🐞 BACKEND REPORTED:  👶 Updating Child Processes:  {"binary":"C:\\Program Files\\Upscayl\\resources\\bin\\upscayl-bin","args":["C:\\Program Files\\Upscayl\\resources\\bin\\upscayl-bin","-i","C:\\!!pictmp\\track.PNG","-o","C:\\!!pictmp\\track_upscayl_4x_realesrgan-x4plus.png","-s","4","-m","C:\\Program Files\\Upscayl\\resources\\models","-n","realesrgan-x4plus","-g","0","-f","png"]}
🐞 BACKEND REPORTED:  🛑 Updating Stopped:  false
🐞 BACKEND REPORTED:  image upscayl:  [0 AMD Radeon RX 7900 XT]  queueC=1[2]  queueG=0[1]  queueT=2[2]
[0 AMD Radeon RX 7900 XT]  bugsbn1=0  bugbilz=0  bugcopc=0  bugihfa=0
[0 AMD Radeon RX 7900 XT]  fp16-p/s/a=1/1/1  int8-p/s/a=1/1/1
[0 AMD Radeon RX 7900 XT]  subgroup=64  basic=1  vote=1  ballot=1  shuffle=1

🚧 UPSCAYL_PROGRESS:  [0 AMD Radeon RX 7900 XT]  queueC=1[2]  queueG=0[1]  queueT=2[2]
[0 AMD Radeon RX 7900 XT]  bugsbn1=0  bugbilz=0  bugcopc=0  bugihfa=0
[0 AMD Radeon RX 7900 XT]  fp16-p/s/a=1/1/1  int8-p/s/a=1/1/1
[0 AMD Radeon RX 7900 XT]  subgroup=64  basic=1  vote=1  ballot=1  shuffle=1

🐞 BACKEND REPORTED:  image upscayl:  vkAllocateMemory failed -13

🔄 Resetting image paths
🚧 UPSCAYL_PROGRESS:  vkAllocateMemory failed -13

🐞 BACKEND REPORTED:  ❌ INVALID GPU OR FAILED
🐞 BACKEND REPORTED:  image upscayl:  vkQueueSubmit failed -2

🔄 Resetting image paths
🚧 UPSCAYL_PROGRESS:  vkQueueSubmit failed -2

🐞 BACKEND REPORTED:  ❌ INVALID GPU OR FAILED
⚙️ Getting saveImageAs from localStorage:  png
⚙️ Getting model from localStorage:  {"label":"General Photo (Real-ESRGAN)","value":"realesrgan-x4plus"}
⚙️ Getting gpuId from localStorage:  0
⚙️ Getting rememberOutputFolder from localStorage:  false
```

**Comment by aaronliu0130** [comment]
Sounds like you ran out of memory

**Comment by NikAWing** [comment]
yeah, but I doubt it with 20GB VRAM :o

**Comment by NayamAmarshe** [comment]
Try reinstalling drivers after uninstalling them

**Comment by NikAWing** [comment]
that did not work though :/
(but I have this AMD service error about "changed hardware settings" after every reboot)
Just to be complete sure that this is not the problem: I updated Upscayl from 2.9.2 to 2.9.8 by extracting the zip file into the program folder instead of using the installer. This should be alright?

Maybe someone else has a new(er) AMD card and might know how to fix?

**Comment by aaronliu0130** [comment]
Yeah, the install method shouldn't affect anything.

I in fact have a much older card (RX 580). Dunno what's causing the issue. Perhaps the input image is so large it takes up... 20GB? 🧐

**Comment by NayamAmarshe** [comment]
> Yeah, the install method shouldn't affect anything.
> 
> I in fact have a much older card (RX 580). Dunno what's causing the issue. Perhaps the input image is so large it takes up... 20GB? 🧐

I don't believe it's a memory issue.
Try this @NikAWing: I WIN+CTRL+SHIFT+B, it is a hotkey to reset the video driver on Window.

**Comment by NikAWing** [comment]
that also didn't work ... whatever I try I get the vkAllocateMemory failed -13 error :o

**Comment by NikAWing** [comment]
no changes with 2.9.9 either hmm

**Comment by aaronliu0130** [comment]
We have no idea, unfortunately

**Comment by NikAWing** [comment]
maybe someone else who's reading here knows what's up.

Just an idea that came up right now: resizable BAR could not be the culprit? I never used it before and activated it since I have the new video card. AMD software says "res. BAR is using 256 MB right now"

**Comment by UnKnoWn-Consortium** [comment]
FYI I have upscayl running on an external RX 7600M XT on my desktop rig via Thunderbolt and it runs fine. The desktop itself has a RX 6900 XT installed and multi-GPU also works. 

Just curious have you tried a clean reinstall of the AMD driver suite, say, with DDU?

**Comment by UnKnoWn-Consortium** [comment]
btw -13 is `VK_ERROR_UNKNOWN` and -2 is `VK_ERROR_OUT_OF_DEVICE_MEMORY`. 

FYR https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkResult.html

**Comment by UnKnoWn-Consortium** [comment]
btw -13 is `VK_ERROR_UNKNOWN` and -2 is `VK_ERROR_OUT_OF_DEVICE_MEMORY`. So basically it is a cascading error. Failure to allocate the device memory due to an unknown error retarded the subsequent enqueuing of buffers/commands.

FYR https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkResult.html

**Comment by NikAWing** [comment]
ok, I did that 3 times, but not with DDU.
I just was about to try DDU, downloaded it, downloaded the AMD driver again but Win10 wanted to install some updates first.
So after updating I tried upscayl again and this time it was not complaining at all, it actually upscaled really fast :D

I have no idea what it was, I rebooted before, I also played around with stable diffusion (without being too happy about its speed), upscayl never wanted to do its work. But it does now after a fr*kken windows update ... lol

Thanks, everyone! I'll close this issue with this comment :)
